src/common/transformations/src/transformations/common_optimizations/split_squeeze_concat_fusion.cpp

```diff
@@ -120,3 +107,65 @@ ov::pass::SplitSqueezeConcatFusion::SplitSqueezeConcatFusion() {
     auto m = std::make_shared<ov::pass::pattern::Matcher>(concat_pattern, matcher_name);
     register_matcher(m, callback);
 }
+
+bool is_axis_squeezed_by_node(const std::shared_ptr<ov::Node>& squeeze_node, int64_t axis) {
+    const auto& input_shape = squeeze_node->get_input_partial_shape(0);
+    const auto& output_shape = squeeze_node->get_output_partial_shape(0);
+    if (input_shape.rank().is_dynamic() || output_shape.rank().is_dynamic())
+        return false;
+
+    auto input_rank = input_shape.rank().get_length();
+    auto output_rank = output_shape.rank().get_length();
+    // check if output_rank == input_rank - 1
+    // to make sure the node actually squeezes a dimension
+    if (input_rank != output_rank + 1)
+        return false;
+
+    // check if squeezed dimension equals to 1
+    if (input_shape[axis].is_dynamic() || input_shape[axis] != 1)
+        return false;
+
+    if (ov::is_type<ov::op::v1::Reshape>(squeeze_node)) {
```

Comment:
it looks like we need to check 2nd input to Reshape operation because it can contain these values:
![image](https://github.com/openvinotoolkit/openvino/assets/21181928/3943ace0-63c3-467d-9774-8e10d21c050e)
we might break reshapability in some cases

Reply:
fixed